kernel/src/main/scala/org/apache/toree/kernel/api/Kernel.scala

```diff
@@ -414,13 +417,15 @@ class Kernel (
           Await.result(sessionFuture, Duration(100, TimeUnit.MILLISECONDS))
         } catch {
           case timeout: TimeoutException =>
-            // getting the session is taking a long time, so assume that Spark
-            // is starting and print a message
-            display.content(
-              MIMEType.PlainText, "Waiting for a Spark session to start...")
+            // in cluster mode, the sparkContext is forced to initialize
+            if (SparkUtils.isSparkClusterMode(defaultSparkConf) == false) {
```

Comment:
Instead of preventing the message from being sent, I think this should update the logic so that this uses the second case. That case just creates a new context without the futures or Await calls.